team/tests/org.eclipse.compare.tests/src/org/eclipse/compare/tests/DocLineComparatorTest.java

```diff
@@ -258,15 +259,67 @@ public boolean canCacheFilteredRegions() {
 		for (int i = 0; i < docs.length; i++)
 			for (int j = i + 1; j < docs.length; j++)
 				for (ICompareFilter[] filter : filters) {
-					l = new DocLineComparator(docs[i], null, false, filter, 'L');
-					r = new DocLineComparator(docs[j], null, false, filter, 'R');
+					l = new DocLineComparator(docs[i], null, false, filter, 'L', Optional.empty());
+					r = new DocLineComparator(docs[j], null, false, filter, 'R', Optional.empty());
 					Assert.assertFalse(l.rangesEqual(0, r, 0));
-					l = new DocLineComparator(docs[i], null, true, filter, 'L');
-					r = new DocLineComparator(docs[j], null, true, filter, 'R');
+					l = new DocLineComparator(docs[i], null, true, filter, 'L', Optional.empty());
+					r = new DocLineComparator(docs[j], null, true, filter, 'R', Optional.empty());
 					Assert.assertTrue(l.rangesEqual(0, r, 0));
 				}
 	}
 
+	@Test
+	public void whiteSpaceInStringLiteralWithEmptyIgnoreWhitespaceContributor() {
+		DocLineComparator left = new DocLineComparator(new Document("str = \"Hello World\""), null, true, null, 'L', Optional.empty());
+		DocLineComparator right = new DocLineComparator(new Document("str = \"HelloWorld\""), null, true, null, 'R', Optional.empty());
+		Assert.assertTrue(left.rangesEqual(0, right, 0));
+		Assert.assertTrue(right.rangesEqual(0, left, 0));
+	}
+
+	@Test
+	public void whiteSpaceInStringLiteralWithSimpleIgnoreWhitespaceContributor() {
```

Comment:
The method name does not state the expectation.
Maybe

```suggestion
	public void simpleWhitespaceContributorSupplied_whitespaceInStringLiteralIgnored() {
```

is a better name.

Reply:
changed with 9376205, correct name is simpleWhitespaceContributorSupplied_whitespaceInStringLiteralNotIgnored